### DISCUSSION TITLE
WASasquatch nodes don't import

### DISCUSSION THREAD

**Discussion by bobpuffer1** [WASasquatch/was-node-suite-comfyui#176] [Q&A]
Keep getting the following error:

ImportError: Numba needs NumPy 1.24 or less

When I downgrade NumPy ComfyUI doesn't load
Any ideas?

**Comment by WASasquatch** [comment]
I use 1.24 so not sure what that's about or why it would be involved with the startup process (which it isn't). You probably need to uninstall all that and install properly thoroughly the requirements.txt file or try the install.bat

**Reply by bobpuffer1** [reply]
I tried reinstalling using git clone and install.bat and I ran into three problems:

1. the install.bat file doesn't like that I have a space in my path AKA \Users\Bob Puffer. I had to edit to use an absolute path. That may be something you wanna correct for future versions. I would write an issue if you want me to?
2. I get this warning basically telling me it's not going to install numpy. I looked at the requirements file and it looked like it was ok.

WARNING: Ignoring invalid distribution -umpy (c:\users\bob puffer\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages)

3. I get this error at the end which basically means to me that it's not going to let me use WAS. Installing as an administrator doesn't change anything.

ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'C:\\Users\\Bob Puffer\\AppData\\Local\\Temp\\pip-uninstall-yhre4ine\\libopenblas64__v0.3.21-gcc_10_3_0.dll'
Check the permissions.

**Reply by WASasquatch** [reply]
You are using system python. Are you using the portable which has it's own python distribution in /python_embedded/python.exe ?

If you have portable, installing to System won't install anything to the ComfyUI portable. You need to install to it like:

`"C:\ComfyUI_windows_portable\python_embeded\python.exe" -s -m pip install -r "C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\requirements.txt"`

**Reply by bobpuffer1** [reply]
Yes I'm using the system Python. I apologize for not reading your instructions more carefully. I've gone Through the pip install and I get a couple warnings about opencv and numPy but that's it. however, when I start comfy ui I get an error that prevents me from running:

Traceback (most recent call last):
  File "C:\Users\Bob Puffer\ComfyUI\main.py", line 66, in <module>
    import comfy.utils
  File "C:\Users\Bob Puffer\ComfyUI\comfy\utils.py", line 1, in <module>
    import torch
  File "C:\Users\Bob Puffer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\torch\__init__.py", line 934, in <module>
    from .storage import _StorageBase, TypedStorage, _LegacyStorage, UntypedStorage, _warn_typed_storage_removal
  File "C:\Users\Bob Puffer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\torch\storage.py", line 12, in <module>
    import numpy as np
  File "C:\Users\Bob Puffer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\numpy\__init__.py", line 240, in <module>
    __all__.remove('min')
ValueError: list.remove(x): x not in list

I have no other python processes running
thanks

**Reply by bobpuffer1** [reply]
I should mention that if I remove WAS I still get the error

**Comment by bobpuffer1** [comment]
I upgraded numpy and comfy ui loads now

**Reply by WASasquatch** [reply]
May have been left over file that didn't get overwritten and had junk that current install didn't understand

**Comment by bobpuffer1** [comment]
I can get ConfyUI to work with numPy 1.24.0 but now I get WAS stalled on:

WAS Node Suite: Installing `opencv-python-headless` ...
Collecting opencv-python-headless[ffmpeg]
  Using cached opencv_python_headless-4.8.0.76-cp37-abi3-win_amd64.whl (38.0 MB)
_**WARNING: opencv-python-headless 4.8.0.76 does not provide the extra 'ffmpeg'
Collecting numpy>=1.17.0**_
  Using cached numpy-1.25.2-cp310-cp310-win_amd64.whl (15.6 MB)
Installing collected packages: numpy, opencv-python-headless
**_ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'C:\\Users\\Bob Puffer\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python310\\site-packages\\numpy\\core\\_multiarray_tests.cp310-win_amd64.pyd'
Check the permissions._**


[notice] A new release of pip is available: 23.0.1 -> 23.2.1
[notice] To update, run: C:\Users\Bob Puffer\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip
Traceback (most recent call last):
  File "C:\Users\Bob Puffer\ComfyUI\nodes.py", line 1693, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\Users\Bob Puffer\ComfyUI\custom_nodes\was-node-suite-comfyui\__init__.py", line 1, in <module>
    from .WAS_Node_Suite import NODE_CLASS_MAPPINGS
  File "C:\Users\Bob Puffer\ComfyUI\custom_nodes\was-node-suite-comfyui\WAS_Node_Suite.py", line 13283, in <module>
    subprocess.check_call([sys.executable, '-s', '-m', 'pip', 'install', 'opencv-python-headless[ffmpeg]'])
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['C:\\Users\\Bob Puffer\\AppData\\Local\\Microsoft\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\\python.exe', '-s', '-m', 'pip', 'install', 'opencv-python-headless[ffmpeg]']' returned non-zero exit status 1.

it doesn't matter if I run it as administrator I still get the permissions issue for the temp directory

**Reply by WASasquatch** [reply]
Yes cause it can't detect opencv python because it was never installed or installed properly to ComfyUI portable.

Using system python means you don't have all the various build tools and need to install them yourself and build dependencies. That will be a headache. You should only be using the stand alone portable version and that's the only version I support.

**Comment by bobpuffer1** [comment]
Well... that was an important piece I must accidentally have skipped over. it's amazing how well things work if I follow instructions. thank you for your assistance I have WAS enabled and am running ComfyUI. there's a reason I didn't use standalone. I got some errors that I didn't get with the system python. well I'm going to go down this road now.

**Reply by WASasquatch** [reply]
I may be able to help with those errors. Though I don't know if you know, there is a Matrix Chatroom on the Element chat app for ComfUI dev and talk, as well as sharing. It's a invaluable resource for sure. I suggest joining in. We can help you there with any issues you may have. 

https://app.element.io/#/room/%23comfyui_space%3Amatrix.org

**Comment by bobpuffer1** [comment]
thanks for the tip on the chat room I joined up
